From f5ec5a0737afd28416a5b45b4ffe0465920baec7 Mon Sep 17 00:00:00 2001
From: Bjorn Helgaas <bhelgaas@google.com>
Date: Tue, 30 Jan 2018 12:12:43 -0600
Subject: [PATCH] PCI/DPC: Push dpc->rp_pio_status assignment into dpc_rp_pio_get_info()
Git-commit: f5ec5a0737afd28416a5b45b4ffe0465920baec7
Patch-mainline: v4.16-rc1
References: FATE#325195

Move the dpc->rp_pio_status assignment into dpc_rp_pio_get_info() since
that's where we read rp_pio->status anway.  No functional change intended.

Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
Reviewed-by: Keith Busch <keith.busch@intel.com>
Reviewed-by: Sinan Kaya <okaya@codeaurora.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/pci/pcie/pcie-dpc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/pcie/pcie-dpc.c b/drivers/pci/pcie/pcie-dpc.c
index fb7ddbe96811..7b96169a808c 100644
--- a/drivers/pci/pcie/pcie-dpc.c
+++ b/drivers/pci/pcie/pcie-dpc.c
@@ -158,6 +158,8 @@ static void dpc_rp_pio_get_info(struct dpc_dev *dpc,
 	dev_err(dev, "rp_pio_status: %#010x, rp_pio_mask: %#010x\n",
 		rp_pio->status, rp_pio->mask);
 
+	dpc->rp_pio_status = rp_pio->status;
+
 	pci_read_config_dword(pdev, cap + PCI_EXP_DPC_RP_PIO_SEVERITY,
 			      &rp_pio->severity);
 	pci_read_config_dword(pdev, cap + PCI_EXP_DPC_RP_PIO_SYSERROR,
@@ -213,7 +215,6 @@ static void dpc_process_rp_pio_error(struct dpc_dev *dpc)
 
 	dpc_rp_pio_get_info(dpc, &rp_pio_regs);
 
-	dpc->rp_pio_status = rp_pio_regs.status;
 }
 
 static irqreturn_t dpc_irq(int irq, void *context)
-- 
2.19.2

